src/js/typescript.js

```diff
@@ -81,9 +81,11 @@ export default class TypeScriptCompiler extends SimpleCompilerBase {
   addHotModuleLoadingRegistration(sourceCode, fileName, exports) {
     if (exports.length < 1) return sourceCode;
 
-    let registrations = exports.map(x => 
-      `__REACT_HOT_LOADER__.register(${x}, "${x}", __FILENAME__);\n`
-    );
+    let registrations = exports.map(x => {
+      let id = `${x}` == 'default' ? "_default" : `${x}`
```

Comment:
It looks like that to be the case at least with the ***react-hot-loader*** package

Default is interpolated as _default. Though it looks like it's some sort of package specific constant that they use?

https://github.com/gaearon/react-hot-loader/blob/9146001a9e8f076ad36669daa3c2bcdfb1631905/test/babel/fixtures/bindings/expected.js